From: uazo <uazo@users.noreply.github.com>
Date: Mon, 2 Sep 2024 07:03:38 +0000
Subject: Add a GN flag to disable trybot verification

Adds flag in GN to deactivate the check performed by some google build bots.
The flag is deactivated when generating the system webview with
skip_secondary_abi_for_cq=true in arm64.

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 chrome/android/expectations/expectations.gni | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/chrome/android/expectations/expectations.gni b/chrome/android/expectations/expectations.gni
--- a/chrome/android/expectations/expectations.gni
+++ b/chrome/android/expectations/expectations.gni
@@ -5,8 +5,12 @@
 import("//build/config/android/config.gni")
 import("//build/config/coverage/coverage.gni")
 
+declare_args() {
+    enable_trybot_verification = true
+}
+
 # Chosen to match android-binary-size trybot.
-enable_libs_and_assets_verification =
+enable_libs_and_assets_verification = enable_trybot_verification &&
     !enable_chrome_android_internal && !is_component_build &&
     !use_jacoco_coverage && is_official_build &&
     (target_cpu == "arm64" && android_channel == "stable")
@@ -15,6 +19,6 @@ enable_libs_and_assets_verification =
 # use_jacoco_coverage: Adds proguard flags.
 # android_channel: Manifest entries vary based on channel.
 # target_cpu: android:versionCode varies.
-enable_manifest_verification =
+enable_manifest_verification = enable_trybot_verification &&
     !is_java_debug && !enable_chrome_android_internal && !use_jacoco_coverage &&
     (target_cpu == "arm64" && android_channel == "stable")
--
